src/hooks/useDialogComponent.ts

```diff
@@ -0,0 +1,127 @@
+  DialogTrigger,
+} from "@radix-ui/react-dialog";
+
+interface DialogComponentProps {
+  dialogType: "dialog" | "alert";
+  type: "fullScreen" | "card";
+}
+type DialogComponentType<T> =
+  | React.FC<T>
+  | React.ForwardRefExoticComponent<
+      Omit<T & React.RefAttributes<unknown>, "ref">
+    >;
+
+type ComponentWithDisplayName = {
+  ({ className, ...props }: React.HTMLAttributes<HTMLDivElement>): JSX.Element;
+  displayName: string;
+};
+
+type DialogComponents = {
+  DialogComponent: DialogComponentType<AlertDialogProps | DialogProps>;
+  DialogTriggerComponent: DialogComponentType<AlertDialogTriggerProps>;
+  DialogContentComponent: DialogComponentType<
+    AlertDialogContentProps | DialogContentProps
+  >;
+  DialogHeaderComponent:
+    | ComponentWithDisplayName
+    | (({ children, className }: InlineDialogProps) => JSX.Element);
+  DialogTitleComponent: DialogComponentType<
+    AlertDialogTitleProps | DialogTitleProps
+  >;
+  DialogDescriptionComponent: DialogComponentType<
+    AlertDialogDescriptionProps | DialogDescriptionProps
+  >;
+  DialogFooterComponent:
+    | ComponentWithDisplayName
+    | (({ children, className }: InlineDialogProps) => JSX.Element);
+  DialogActionComponent: DialogComponentType<AlertDialogActionProps>;
+  DialogCancelComponent: DialogComponentType<AlertDialogCancelProps>;
+};
+
+export default function useDialogComponent({
+  dialogType,
+  type,
+}: DialogComponentProps): DialogComponents {
+  const isFullScreen = type === "fullScreen";
+
+  if (dialogType === "alert") {
+    return {
+      DialogComponent: AlertDialog,
+      DialogTriggerComponent: AlertDialogTrigger,
+      DialogContentComponent: AlertDialogContent,
+      DialogHeaderComponent: isFullScreen
+        ? AlertDialogHeader
+        : InlineDialogHeader,
+      DialogTitleComponent: isFullScreen ? AlertDialogTitle : InlineDialogTitle,
+      DialogDescriptionComponent: isFullScreen
+        ? AlertDialogDescription
+        : InlineDialogDescription,
+      DialogFooterComponent: isFullScreen
+        ? AlertDialogFooter
+        : InlineDialogFooter,
+      DialogActionComponent: isFullScreen
+        ? AlertDialogAction
+        : InlineDialogAction,
+      DialogCancelComponent: isFullScreen
+        ? AlertDialogCancel
+        : InlineDialogCancel,
+    };
+  }
+
+  if (dialogType === "dialog") {
+    return {
+      DialogComponent: Dialog,
+      DialogTriggerComponent: DialogTrigger,
+      DialogContentComponent: DialogContent,
+      DialogHeaderComponent: isFullScreen ? DialogHeader : InlineDialogHeader,
+      DialogTitleComponent: isFullScreen ? DialogTitle : InlineDialogTitle,
+      DialogDescriptionComponent: isFullScreen
+        ? DialogDescription
+        : InlineDialogDescription,
+      DialogFooterComponent: isFullScreen ? DialogFooter : InlineDialogFooter,
+      DialogActionComponent: InlineDialogAction,
+      DialogCancelComponent: InlineDialogCancel,
+    };
+  }
+
+  return {} as DialogComponents;
```

Comment:
why have to return {} ?

Reply:
I added `return {} as DialogCOmponents;` to match the type of return value in case TypeScript does not pass all if conditional statements. Since the type itself alert or dialog, it would be better to modify the if conditional statement and remove `return {} as DialogCOmponents;`